airflow/jobs/base_job_runner.py

```diff
@@ -17,22 +17,27 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Generic, TypeVar
 
 from airflow.utils.session import NEW_SESSION, provide_session
 
 if TYPE_CHECKING:
     from sqlalchemy.orm import Session
 
     from airflow.jobs.job import Job
+    from airflow.serialization.pydantic.job import JobPydantic
 
+T = TypeVar("T", bound="Job | JobPydantic")
 
-class BaseJobRunner:
+JobOrJobPydantic = TypeVar("JobOrJobPydantic", bound="Job | JobPydantic")
```

Comment:
Only one of this and `T` needs to exist.

Reply:
Even nicer!